array_test.go

```diff
@@ -604,78 +604,245 @@ func TestArray_IsEqualUnordered(t *testing.T) {
 		value.chain.assertFailed(t)
 		value.chain.clearFailed()
 	})
+
+	t.Run("canonization", func(t *testing.T) {
+		type (
+			myArray []interface{}
+			myInt   int
+		)
+
+		value := NewArray(reporter, []interface{}{123, 456, "foo"})
+		duplicateValue := NewArray(reporter, []interface{}{123, 123, "foo"})
+
+		assert.Equal(t, []interface{}{123.0, 456.0, "foo"}, value.Raw())
+
+		value.IsEqualUnordered(myArray{myInt(456), 123.0, "foo"})
+		value.chain.assertNotFailed(t)
+		value.chain.clearFailed()
+
+		value.NotEqualUnordered(myArray{myInt(456), 123.0, "foo"})
+		value.chain.assertFailed(t)
+		value.chain.clearFailed()
+
+		value.IsEqualUnordered(myArray{456.0, myInt(123.0), "foo"})
+		value.chain.assertNotFailed(t)
+		value.chain.clearFailed()
+
+		value.NotEqualUnordered(myArray{456.0, myInt(123.0), "foo"})
+		value.chain.assertFailed(t)
+		value.chain.clearFailed()
```

Comment:
This check is like previous one, but with changed order. We can assume that the ordering part of IsEqualUnordered is already tested and works well, and remove this check from canonization subtest. We need to test only for different types, but we don't need to test all permutations of them.

Reply:
Same for TestArray_ContainsOnly/canonization - the number of checks can be reduced I think.

Reply:
I've reduced the tests for these 
i.e `TestArray_IsEqualUnordered` and `TestArray_ContainsOnly`